### DISCUSSION TITLE
Data tables component to consume external data - JSON format?

### DISCUSSION THREAD

**Discussion by Kugeleis** [vuetifyjs/vuetify#13065] [Q&A]
I try to use the v-data-table constructor. The [examples](https://vuetifyjs.com/en/components/data-tables/#usage) all are with local data. But how can I display external data (e. g. JSON data obtained via fetch?

**Comment by nekosaur** [comment]
This example https://vuetifyjs.com/en/components/data-tables/#server-side-paginate-and-sort shows how you can work with server data

**Reply by Kugeleis** [reply]
Thanks. So I still have to build the header structure myself deviding the data delivered if it contains the whole table information?

**Reply by nekosaur** [reply]
You need to provide `headers` prop yes. I'm not exactly sure what you mean by dividing the data. Perhaps an example JSON response could help illustrate your question.

**Reply by Kugeleis** [reply]
I actually get data from a python pandas dataframe export. It has [many options ](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_json.html)but none fits as plug'n play to this vuetify implementation.
A JSON can come like this: `{"columns":["A","B","C"],"index":[0,1,2],"data":[[25,59,28],[61,28,91],[8,70,47]]}` or even a ready made HTML table:
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>A</th>
      <th>B</th>
      <th>C</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>25</td>
      <td>59</td>
      <td>28</td>
    </tr>
    <tr>
      <th>1</th>
      <td>61</td>
      <td>28</td>
      <td>91</td>
    </tr>
    <tr>
      <th>2</th>
      <td>8</td>
      <td>70</td>
      <td>47</td>
    </tr>
  </tbody>
</table>
I attached all options:
[tabledata.zip](https://github.com/vuetifyjs/vuetify/files/5936967/tabledata.zip)

**Reply by nekosaur** [reply]
You could do something like this for example https://codepen.io/nekosaur/pen/PobNGrJ?editors=1010

**Reply by Kugeleis** [reply]
This looks great and is dynamic. Thanks a lot!
